pkg/didcomm/common/service/destination_default.go

```diff
@@ -30,8 +31,12 @@ func CreateDestination(didDoc *diddoc.Doc) (*Destination, error) {
 		return nil, fmt.Errorf("create destination: no recipient keys on didcomm service block in diddoc: %+v", didDoc)
 	}
 
-	// TODO ensure recipient keys are did:key's
-	//  https://github.com/hyperledger/aries-framework-go/issues/1604
+	for i, k := range didCommService.RecipientKeys {
+		if !strings.HasPrefix(k, "did:") {
```

Comment:
This is testing against prefix `did:`, so I guess we're still accepting did-url key references, right?

Reply:
This is a check to ensure RecipientKeys are always didkeys

The didCommService is already updated to have RecipientKeys as did keys. This is only a validation as per the removed TODO comment above.

Reply:
> Noticed a few small places where we could refactor to clean up a bit, but those can come later.
> LGTM 👍

Indeed there's lot's of room for improvements.